code/game/antagonist/station/revolutionary/excel_faction.dm

```diff
@@ -81,15 +81,27 @@
 		to_chat(usr, SPAN_NOTICE("The stash has already been summoned by \"[F.stash_holder]\""))
 		return
 
-	var/mob/living/carbon/human/H = usr
-
-	var/obj/item/storage/deferred/stash/sack/stash = new
-
-	new /obj/item/computer_hardware/hard_drive/portable/design(stash)
-	new /obj/item/computer_hardware/hard_drive/portable/design/excelsior/core(stash)
-	new /obj/item/computer_hardware/hard_drive/portable/design/excelsior/weapons(stash)
-	new /obj/item/electronics/circuitboard/excelsiorautolathe(stash)
-	new /obj/item/electronics/circuitboard/excelsior_teleporter(stash)
-
-	H.put_in_hands(stash)
-	F.stash_holder = H.real_name
\ No newline at end of file
+	if(alert(usr,"Are you sure you want to summon Excelsior stash?","Means of Production","Yes, the time has come","No, not yet") == "Yes, the time has come")
+		F.stash_holder = H.real_name
+		var/text = "<span class='revolution'>\"Comrade [F.stash_holder] has seized the means of production!\"</span>"
```

Comment:
Area, you say...
```suggestion
		var/area/comrade_area = get_area(H)
		var/comrade_location = initial(comrade_area.name)
		var/text = "<span class='revolution'>\"Comrade [H] has seized the means of production in [comrade_location]!\"</span>"
```
`Initial()` thing needed in case you don't want excels to summon stash to the gay bar.